centreon/www/include/monitoring/downtime/listDowntime.php

```diff
@@ -123,116 +133,168 @@
-$downtimesStatement->bindValue(':host', '%' . $host_name . '%', \PDO::PARAM_STR);
-$downtimesStatement->bindValue(':output', '%' . $search_output . '%', \PDO::PARAM_STR);
-$downtimesStatement->bindValue(':author', '%' . $search_author . '%', \PDO::PARAM_STR);
-$downtimesStatement->bindValue(':offset', $num * $limit, \PDO::PARAM_INT);
-$downtimesStatement->bindValue(':limit', $limit, \PDO::PARAM_INT);
-$downtimesStatement->execute();
-
-$rows = $pearDBO->query("SELECT FOUND_ROWS() AS REALTIME")->fetchColumn();
-
-for ($i = 0; $data = $downtimesStatement->fetchRow(); $i++) {
+
+$subQueryConditionSearchEndTime = '';
+if ($canViewAll === false) {
+    $subQueryConditionSearchEndTime = 'AND d.end_time > :end_time ';
+    $bindValues[':end_time'] = [time(), PDO::PARAM_INT];
+}
+
+$serviceQuery = <<<SQL
+    SELECT SQL_CALC_FOUND_ROWS DISTINCT
+        1 AS REALTIME, d.internal_id as internal_downtime_id, d.entry_time, duration,
+        d.author as author_name, d.comment_data, d.fixed as is_fixed, d.start_time as scheduled_start_time,
+        d.end_time as scheduled_end_time, d.started as was_started, d.host_id, d.service_id, h.name as host_name,
+        s.description as service_description {$extraFields}
+    FROM downtimes d
+    INNER JOIN services s
+        ON d.host_id = s.host_id
+        AND d.service_id = s.service_id
+    INNER JOIN hosts h
+        ON h.host_id = s.host_id
+    {$serviceAclSubRequest}
+    WHERE d.type = {$serviceType}
+        {$subQueryConditionCancelled}
+        {$subQueryConditionSearchService}
+        {$subQueryConditionSearchOutput}
+        {$subQueryConditionSearchAuthor}
+        {$subQueryConditionSearchHost}
+        {$subQueryConditionSearchEndTime}
+    SQL;
+
+$hostQuery = <<<SQL
+    SELECT DISTINCT
+        1 AS REALTIME, d.internal_id as internal_downtime_id, d.entry_time, duration,
+        d.author as author_name, d.comment_data, d.fixed as is_fixed, d.start_time as scheduled_start_time,
+        d.end_time as scheduled_end_time, d.started as was_started, d.host_id, d.service_id, h.name as host_name,
+        '' as service_description {$extraFields}
+    FROM downtimes d
+    INNER JOIN hosts h
+        ON d.host_id = h.host_id
+    {$hostAclSubRequest}
+    WHERE d.type = {$hostType}
+        {$subQueryConditionCancelled}
+        {$subQueryConditionSearchOutput}
+        {$subQueryConditionSearchAuthor}
+        {$subQueryConditionSearchHost}
+        {$subQueryConditionSearchEndTime}
+    SQL;
+
+$unionQuery = <<<SQL
+    ({$serviceQuery})
+    UNION
+    ({$hostQuery})
+    ORDER BY scheduled_start_time DESC
+    LIMIT :offset, :limit
+    SQL;
+
+$downtimesStatement = $pearDBO->prepareQuery($unionQuery);
+$pearDBO->executePreparedQuery($downtimesStatement, $bindValues, true);
+
+$rows = $pearDBO->executeQuery('SELECT FOUND_ROWS() AS REALTIME')->fetchColumn();
+
+for ($i = 0; $data = $downtimesStatement->fetch(); $i++) {
```

Comment:
To use new methods of centreondb ;) 
```suggestion
$downtimesStatement = $pearDBO->prepareQuery($unionQuery);
$pearDBO->executePreparedQuery($downtimesStatement, $bindValues, true);

$rows = $pearDBO->fetchColumn($pearDBO->executeQuery('SELECT FOUND_ROWS() AS REALTIME'));

for ($i = 0; $data = $pearDBO->fetch($downtimesStatement); $i++) {
```